abdm_integrator/hiu/views/consents.py

```diff
@@ -0,0 +1,225 @@
+        payload['resp'] = {'requestId': self.request_data['requestId']}
+        ABDMRequestHelper().gateway_post(HIUGatewayAPIPath.CONSENT_REQUEST_ON_NOTIFY, payload)
+        return payload['requestId']
+
+
+class GatewayConsentRequestOnFetch(HIUGatewayBaseView):
+
+    def post(self, request, format=None):
+        GatewayConsentRequestOnFetchSerializer(data=request.data).is_valid(raise_exception=True)
+        self.process_request(request.data)
+        return Response(status=HTTP_202_ACCEPTED)
+
+    @transaction.atomic
+    def process_request(self, request_data):
+        consent_artefact = ConsentArtefact.objects.get(gateway_request_id=request_data['resp']['requestId'])
+        if request_data.get('consent'):
+            consent_artefact.details = request_data['consent']['consentDetail']
+            consent_artefact.fetch_status = ArtefactFetchStatus.RECEIVED
+            if consent_artefact.consent_request.artefacts.filter(details__isnull=False).count() == 0:
+                self.update_consent_request_from_artefact(consent_artefact)
+        elif request_data.get('error'):
+            consent_artefact.fetch_status = ArtefactFetchStatus.ERROR
+            consent_artefact.error = request_data['error']
+        consent_artefact.save()
+
+    def update_consent_request_from_artefact(self, consent_artefact):
+        consent_request = consent_artefact.consent_request
+        consent_request.update_user_amendable_details(consent_artefact.details['permission'],
+                                                      consent_artefact.details['hiTypes'])
+
+
+class ConsentFetch(HIUBaseView, viewsets.ReadOnlyModelViewSet):
+    serializer_class = ConsentRequestSerializer
+    pagination_class = APIResultsSetPagination
+
+    def get_queryset(self):
+        queryset = ConsentRequest.objects.all().order_by('-date_created')
+        request_params = self.request.query_params
+        if request_params.get('abha_id'):
+            queryset = queryset.filter(details__patient__id=request_params['abha_id'])
+        if request_params.get('status'):
+            queryset = queryset.filter(status=request_params['status'])
+        if request_params.get('from_date'):
+            queryset = queryset.filter(health_info_to_date__date__gte=request_params['from_date'])
+        if request_params.get('to_date'):
+            queryset = queryset.filter(health_info_from_date__date__lte=request_params['to_date'])
+        if request_params.get('search'):
+            queryset = queryset.filter(Q(status__icontains=request_params['search']) |
+                                       Q(health_info_types__icontains=request_params['search']))
+        return queryset
+
+
+class ConsentArtefactFetch(HIUBaseView, viewsets.ReadOnlyModelViewSet):
+    serializer_class = ConsentArtefactSerializer
+    pagination_class = APIResultsSetPagination
+
+    def get_queryset(self):
+        queryset = ConsentArtefact.objects.all().order_by('-date_created')
```

Comment:
I think we are better off moving `all().order_by` to an else condition after line 224

Reply:
Checking if any specific reason for this ? 
We also need order by for the 'list' action.

Reply:
I think applying all() and then applying filters doesn't sound right. The order can be applied at the end as well?

Reply:
Got it.  That works . Addressed in 4173ded89fdbdf33077e13ce128fc9627821acda.
_(Got rid of `all()` as not needed for detail or list action)_

Also. I think the earlier version is fine as well as it does not make any difference for actual query because of lazy evaluation.